diff --git a/configure.ac b/configure.ac
index cb670a2a14..154ec88d4a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -673,7 +673,7 @@ AC_SUBST(InstallNameToolCmd)
 # versions of LLVM simultaneously, but that stopped working around
 # 3.5/3.6 release of LLVM.
 LlvmMinVersion=9  # inclusive
-LlvmMaxVersion=13 # not inclusive
+LlvmMaxVersion=14 # not inclusive
 AC_SUBST([LlvmMinVersion])
 AC_SUBST([LlvmMaxVersion])
 sUPPORTED_LLVM_VERSION_MIN=$(echo \($LlvmMinVersion\) | sed 's/\./,/')
